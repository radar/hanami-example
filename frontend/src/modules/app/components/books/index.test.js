import React from 'react';
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import { Books } from './index.js'

const RouterContext = React.createContext('router');

test('Books', () => {
  const data = {
    books: [
      {
        id: 1,
        title: "Exploding Rails",
        permalink: "exploding-rails",
        blurb: "Explode your Rails applications"
      }
    ]
  }
  const component = renderer.create(
    <MemoryRouter>
      <Books data={data} />
    </MemoryRouter>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
