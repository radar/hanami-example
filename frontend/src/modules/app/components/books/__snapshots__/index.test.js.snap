// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books 1`] = `
<div
  className="row"
>
  <div
    className="main col-md-7"
  >
    <h1>
      Your Books
    </h1>
    <div>
      <h2>
        <a
          href="/books/exploding-rails"
          onClick={[Function]}
        >
          Exploding Rails
        </a>
      </h2>
      <span
        className="blurb"
      >
        Explode your Rails applications
      </span>
    </div>
  </div>
</div>
`;
